pkg/docker/config/config.go

```diff
@@ -230,6 +301,16 @@ func RemoveAuthentication(sys *types.SystemContext, registry string) error {
 
 // RemoveAllAuthentication deletes all the credentials stored in auth.json and kernel keyring
 func RemoveAllAuthentication(sys *types.SystemContext) error {
+	credConfigs, err := sysregistriesv2.GetCredConfigs(sys)
+	if err != nil {
+		return err
+	}
+	if len(credConfigs) > 0 {
+		if err := removeAllAuthFromCredConfigs(credConfigs); err == nil {
+			logrus.Debug("Credentials were deleted from credential helper")
+		}
+	}
+
 	return modifyJSON(sys, func(auths *dockerConfigFile) (bool, error) {
 		if enableKeyring {
 			err := removeAllAuthFromKernelKeyring()
```

Comment:
I think it would be consistent to also remove the relevant credentials from the helpers configured in `auths.CredHelpers` below.